core/templates/dev/head/galleries/Gallery.js

```diff
@@ -281,21 +251,34 @@ oppia.controller('Gallery', [
 oppia.controller('SearchBar', [
   '$scope', '$rootScope', '$translate', 'searchService', 'oppiaDebouncer',
   'ExplorationCreationButtonService', 'urlService', 'CATEGORY_LIST',
+  'i18nIdService',
   function(
       $scope, $rootScope, $translate, searchService, oppiaDebouncer,
-      ExplorationCreationButtonService, urlService, CATEGORY_LIST) {
+      ExplorationCreationButtonService, urlService, CATEGORY_LIST,
+      i18nIdService) {
+    $rootScope.$on('$translateChangeSuccess', function() {
+      // This strings must be translated with a filter, generating FoUC for
```

Comment:
Sorry, I don't understand this comment. Do you mean:

    This string must be translated with a filter, otherwise it generates an FoUC for languages other than English

or something else?